cmd/manager/main.go

```diff
@@ -92,15 +93,17 @@ func main() {
 
 	printVersion()
 
-	// Take exclusive instance lock, we need that to make sure that
-	// we don't have more than one working instance of k8s-nmstate
-	handlerLock, err := lockHandler()
-	if err != nil {
-		log.Error(err, "Failed to run lockHandler")
-		os.Exit(1)
+	if !environment.IsOperator() {
+		// Take exclusive instance lock, we need that to make sure that
+		// we don't have more than one working instance of k8s-nmstate
+		handlerLock, err := lockHandler()
+		if err != nil {
+			log.Error(err, "Failed to run lockHandler")
+			os.Exit(1)
+		}
+		log.Info("Successfully took nmstate exclusive lock")
+		defer handlerLock.Unlock()
```

Comment:
We have to call `defer` outside of if statement scope if we don't the handler will be unlock after scope.

The simplest solution is to return a pointer to lock that it's nil if it is an operator so we can do the following:

```golang
defer func() { if handlerLock != nil { handlerLock.Unlock() } }
```

Another option is to create a wrapper around the lock that has the `Unlock` function and to the line below so main function will just have `defer lockWrapper.Unlock()`

Other option is to create a `lockHandlerAndRun(mainFunc func() error) error` that lock handler if is not an operator and then run `mainFunc`

Reply:
Not necessary. Defer runs on func boundaries, not containing block boundaries.

Reply:
Ahh right better!

EDIT: some gotchas on `defer` https://blog.learngoprogramming.com/gotchas-of-defer-in-go-1-8d070894cb01

Reply:
#4 from that same article is what informed me about it. So, is there something in particular that you are concerned about?

Reply:
All good